### PR TITLE
Refactors Map to avoid excessive rendering

### PR DESCRIPTION
**Pull request by madsnedergaard**
## Description

While working on colorblind-mode, I got sucked into the Map.tsx code and before I noticed, I had started reworking some of the logic here 😬 

This PR moves `MapTooltip` out of `Map` and switches to reading/writing from global atoms instead of local state. This will ensure that map the doesn't rerender whenever the cursor moves, and should hopefully make things easier to separate going forward.

**NOTE:** _This might conflict with other PRs, so let's get those merged first!_

### CODE REVIEW

**Comment by VIKTORVAV99** [comment]
> This will ensure that map the doesn't rerender whenever the cursor moves,
Yikes, that's a lot of re-renders. 😅
It's nice to see these PR that keeps increasing our frontend performance getting merged one by one!

File: web/src/features/panels/LeftPanel.tsx [hunk 1]
```diff
@@ -74,7 +75,7 @@ function MobileHeader() {
 }
 
 function OuterPanel({ children }: { children: React.ReactNode }) {
-  const [isOpen, setOpen] = useState(true);
+  const [isOpen, setOpen] = useAtom(leftPanelOpenAtom);
```
Comment (madsnedergaard) [inline web/src/features/panels/LeftPanel.tsx:78]: A small extra thingy, but now we can globally open/close the panel - I just added this to Map, so when clicking a zone it opens the panel (if it's closed) :)

File: web/src/components/CarbonIntensitySquare.tsx [hunk 2]
```diff
@@ -52,7 +52,7 @@ function CarbonIntensitySquare({
           className="mx-auto flex h-[65px] w-[65px] flex-col items-center justify-center rounded-2xl text-md"
         >
           <p className="select-none text-md font-bold" data-test-id="co2-square-value">
-            <animated.div>{number.to((x) => `${Math.round(x) || '?'}g`)}</animated.div>
+            <animated.span>{number.to((x) => `${Math.round(x) || '?'}g`)}</animated.span>
           </p>
```
Comment (madsnedergaard) [inline web/src/components/CarbonIntensitySquare.tsx:56]: Fixes error in console saying a `<div />` cannot be a child of `<p />`

File: web/src/features/map/Map.tsx [hunk 3]
```diff
@@ -232,53 +242,45 @@ export default function MapPage(): ReactElement {
...
+    <Map
+      ref={mapReference}
+      initialViewState={{
+        // TODO: Make these dynamic depending on callerLocation from v6/state
+        latitude: 37.8,
+        longitude: -122.4,
+        zoom: 2,
+      }}
+      interactiveLayerIds={['zones-clickable-layer', 'zones-hoverable-layer']}
+      cursor={hoveredZone ? 'pointer' : 'grab'}
```
Comment (madsnedergaard) [inline web/src/features/map/Map.tsx:254]: Removed the state usage here as it wasn't needed :)

**Review by Kongkille — APPROVED** [review]
The code looks good, 

I guess there'd be other ways to reduce renders but I think most of these atoms are fine to have regardless

**Comment by Kongkille** [comment]
I'll handle the conflicts here

**Comment by madsnedergaard** [comment]
> I'll handle the conflicts here

Thanks! 🙇

**Pull request merged**